ui/app/AppLayouts/CommunitiesPortal/stores/CommunitiesStore.qml

```diff
@@ -85,6 +88,16 @@ QtObject {
         root.communitiesModuleInst.importCommunity(communityKey);
     }
 
+    function requestCommunityInfo(communityKey) {
+        let publicKey = communityKey
+        if (globalUtils.isCompressedPubKey(communityKey)) {
+            publicKey = globalUtils.changeCommunityKeyCompression(communityKey)
+        }
+
+        root.mainModuleInst.setCommunityIdToSpectate(publicKey)
+        root.communitiesModuleInst.requestCommunityInfo(publicKey);
```

Comment:
This looks a bit suspicious. Why does main module know anything about the community to spectate? :thinking:

Reply:
That I don't know. I haven't introduced community spectation but I found that it internally sets a `communityIdToSpectate` when trying to fetch community data from shared community URLs and uses that same ID to later on decide whether or not to perform spectation.

So all I'm doing here is trying to ensure that ID is also set when trying to import via public key.